### DISCUSSION TITLE
How can I assign handleSubmit to a dom element using addEventListener?

### DISCUSSION THREAD

**Discussion by herolindl** [react-hook-form/react-hook-form#4090] [Q&A]
I am passing a ref of a button to my component and I need on that button to add handleSubmit
before it was:
`const {register,handleSubmit,errors} = useForm();`
`<button onClick={handleSubmit(onSubmit)>Submit</button>`
and was working but because of some changes in design the button now is outside the component and I pass a reference of it inside this component where I have to use `handleSubmit` but when I do 
`buttonRef.current.addEventListener("click",handleSubmit(onSubmit));` when I click on the button I get uncaught in promise e.persist is not a function, how can I pass e somehow in my case?

https://codesandbox.io/s/mutable-framework-zf25d?file=/src/App.js

**Comment by bluebill1049** [comment]
share a codesandbox

**Reply by herolindl** [reply]
https://codesandbox.io/s/mutable-framework-zf25d?file=/src/App.js, don't forget that why I am adding by addEventListener is because I am supposed to add it to a buttonRef passed as prop but here I just did like that to demostrate the problem

**Reply by bluebill1049** [reply]
`.addEventListener("click", () => handleSubmit(onSubmit)());`
